workflow/ignite/handlers/best_model_trigger.py

```diff
@@ -1,23 +1,44 @@
+from enum import Enum
+from ignite.engine.engine import CallableEvents
 from ignite.engine import Events
-from workflow.ignite.custom_events import CustomEvents
 import numpy as np
 
 
+class CustomEvents(CallableEvents, Enum):
+    NEW_BEST_MODEL = 'new_best_model'
+
+
 class BestModelTrigger:
-    def __init__(self, engines_to_trigger, metric_to_track):
+    Event = CustomEvents.NEW_BEST_MODEL
+
+    def __init__(self, metric_name, trigger_engines):
         self.best_score = -np.inf
-        self.engines_to_trigger = engines_to_trigger
-        self.metric_to_track = metric_to_track
+        self.trigger_engines = trigger_engines
+        self.metric_name = metric_name
```

Comment:
Curious about the name changes. I feel like the old ones are more descriptive and barely longer, so why the change? To someone reading this code for the first time, I don't think `metric_name` will give them as good of a clue on what that variable is used for. In my opinion variable names should be as descriptive as possible without being overly long.

Reply:
Nothing big. I think `metric_name` is more similar to what it looks like in other places. I think the name of the class is enough to tell us what the metric is used for. `BestModelTrigger(metric_name='accuracy').attach(trainer)`

I also think the naming with "to" was something we are not using anywhere else. What do you think about this?

The second argument is very similar though right? Should probably be optional and be something like `extra_triggered_engines`, I'll change that

Reply:
I also considered changing to the same `model_score_function` that is used elsewhere rather than using a metric

Reply:
Sure, I agree that it's more consistent with our other naming. Let's go with this :)